File: javascript/test/Creational/Factory_Method.js
(function() {
  define(['Creational/Factory_Method/SoupFactoryMethod', 'Creational/Factory_Method/BostonSoupFactoryMethodSubclass', 'Creational/Factory_Method/HonoluluSoupFactoryMethodSubclass'], function(SoupFactoryMethod, BostonSoupFactoryMethodSubclass, HonoluluSoupFactoryMethodSubclass) {
    return describe('Test Factory Method', function() {
      it('Test plain soup factory', function() {
        var sb, sfm;
        sfm = new SoupFactoryMethod();
        sb = sfm.makeSoupBuffet();
        sb.setSoupBuffetName(sfm.makeBuffetName());
        sb.setChickenSoup(sfm.makeChickenSoup());
        sb.setClamChowder(sfm.makeClamChowder());
        sb.setFishChowder(sfm.makeFishChowder());
        sb.setMinnestrone(sfm.makeMinnestrone());
        sb.setPastaFazul(sfm.makePastaFazul());
        sb.setTofuSoup(sfm.makeTofuSoup());
        sb.setVegetableSoup(sfm.makeVegetableSoup());
        return expect("At the " + (sb.getSoupBuffetName()) + " " + (sb.getTodaysSoups())).toEqual("At the Soup Buffet  Today's Soups!  , Chicken Soup: ,ChickenSoup, Clam Chowder: ,ClamChowder, Fish Chowder: ,FishChowder, Minnestrone: ,Minnestrone, Pasta Fazul: ,Pasta Fazul, Tofu Soup: ,Tofu Soup, Vegetable Soup: ,Vegetable Soup");
      });
      it('Test boston soup factory', function() {
        var bsb, bsfm;
        bsfm = new BostonSoupFactoryMethodSubclass();
        bsb = bsfm.makeSoupBuffet();
        bsb.setSoupBuffetName(bsfm.makeBuffetName());
        bsb.setChickenSoup(bsfm.makeChickenSoup());
        bsb.setClamChowder(bsfm.makeClamChowder());
        bsb.setFishChowder(bsfm.makeFishChowder());
        bsb.setMinnestrone(bsfm.makeMinnestrone());
        bsb.setPastaFazul(bsfm.makePastaFazul());
        bsb.setTofuSoup(bsfm.makeTofuSoup());
        bsb.setVegetableSoup(bsfm.makeVegetableSoup());
        return expect("At the  " + (bsb.getSoupBuffetName()) + " " + (bsb.getTodaysSoups())).toEqual("At the  Boston Soup Buffet  Today's Soups!  , Chicken Soup: ,ChickenSoup, Clam Chowder: ,QuahogChowder, Fish Chowder: ,ScrodFishChowder, Minnestrone: ,Minnestrone, Pasta Fazul: ,Pasta Fazul, Tofu Soup: ,Tofu Soup, Vegetable Soup: ,Vegetable Soup");
      });
      return it('Test honolulu soup factory', function() {
        var hsb, hsfm;
        hsfm = new HonoluluSoupFactoryMethodSubclass();
        hsb = hsfm.makeSoupBuffet();
        hsb.setSoupBuffetName(hsfm.makeBuffetName());
        hsb.setChickenSoup(hsfm.makeChickenSoup());
        hsb.setClamChowder(hsfm.makeClamChowder());
        hsb.setFishChowder(hsfm.makeFishChowder());
        hsb.setMinnestrone(hsfm.makeMinnestrone());
        hsb.setPastaFazul(hsfm.makePastaFazul());
        hsb.setTofuSoup(hsfm.makeTofuSoup());
        hsb.setVegetableSoup(hsfm.makeVegetableSoup());
        return expect("At the  " + (hsb.getSoupBuffetName()) + " " + (hsb.getTodaysSoups())).toEqual("At the  Honolulu Soup Buffet  Today's Soups!  , Chicken Soup: ,ChickenSoup, Clam Chowder: ,PacificClamChowder, Fish Chowder: ,OpakapakaFishChowder, Minnestrone: ,Minnestrone, Pasta Fazul: ,Pasta Fazul, Tofu Soup: ,Tofu Soup, Vegetable Soup: ,Vegetable Soup");
      });
    });
  });

}).call(this);

File: scala/test/Creational/FactoryMethod.scala
//TestSoupFactoryMethod - testing the factory method

package test.Creational;

import static org.junit.Assert.assertEquals;

import org.junit.Test;
import org.junit.Ignore;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import Creational.Factory_Method.*;
import common.*;



public class FactoryMethod {         

	public void thetester( SoupFactoryMethod sfm, String msg ,String expected)
	{
       SoupBuffet soupBuffet = sfm.makeSoupBuffet();
       soupBuffet.setSoupBuffetName( sfm.makeBuffetName());
       soupBuffet.setChickenSoup( sfm.makeChickenSoup());
       soupBuffet.setClamChowder( sfm.makeClamChowder()); 
       soupBuffet.setFishChowder( sfm.makeFishChowder());
       soupBuffet.setMinnestrone( sfm.makeMinnestrone());
       soupBuffet.setPastaFazul( sfm.makePastaFazul()); 
       soupBuffet.setTofuSoup( sfm.makeTofuSoup());
       soupBuffet.setVegetableSoup( sfm.makeVegetableSoup()); 
       org.junit.Assert.assertEquals(msg,
			expected,
			 ("At the  " + soupBuffet.getSoupBuffetName() + soupBuffet.getTodaysSoups())
		);
	}
       

	@Test
   public void base_sfm() {
		thetester( new SoupFactoryMethod(), "Base Soup Factory Method",
			"At the  Soup Buffet Today's Soups!   Chicken Soup: ChickenSoup Clam Chowder: ClamChowder Fish Chowder: FishChowder Minnestrone: Minestrone Pasta Fazul: Pasta Fazul Tofu Soup: Tofu Soup Vegetable Soup: Vegetable Soup"
		);
	}
       
	@Test
   public void boston_sfm() {
		thetester( new BostonSoupFactoryMethodSubclass(), "Boston Soup Factory Method",
			"At the  Boston Soup Buffet Today's Soups!   Chicken Soup: ChickenSoup Clam Chowder: QuahogChowder Fish Chowder: ScrodFishChowder Minnestrone: Minestrone Pasta Fazul: Pasta Fazul Tofu Soup: Tofu Soup Vegetable Soup: Vegetable Soup"
		);
	}
       
	@Test
   public void hono_sfm() {
		thetester( new HonoluluSoupFactoryMethodSubclass(), "Honolulu Soup Factory Method",
			"At the  Honolulu Soup Buffet Today's Soups!   Chicken Soup: ChickenSoup Clam Chowder: PacificClamChowder Fish Chowder: OpakapakaFishChowder Minnestrone: Minestrone Pasta Fazul: Pasta Fazul Tofu Soup: Tofu Soup Vegetable Soup: Vegetable Soup"
		);
	}
}      
